invenio-assets: CI error when updating to v1.3.1

When updating to v1.3.1 (currently using 1.3.0) of invenio-assets, the **setup npm and build assets** section of the CI fails with the error **Error: Cannot find module 'ajv/dist/compile/codegen'**.

--- a/INSTALL.rst
+++ b/INSTALL.rst
@@ -118,7 +118,7 @@ You may want to set this automatically in your bash or zsh profile.
 
 .. code-block:: console
 
-   (venv)$ export FLASK_ENV=development
+   (venv)$ export FLASK_DEBUG=1
 
 Use of config_local.py
 ----------------------

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -22,7 +22,7 @@ services:
     - "APP_SQLALCHEMY_DATABASE_URI=postgresql://hepdata:hepdata@db/hepdata"
     - "SAUCE_USERNAME=${SAUCE_USERNAME}"
     - "SAUCE_ACCESS_KEY=${SAUCE_ACCESS_KEY}"
-    - "FLASK_ENV=development"
+    - "FLASK_DEBUG=1"
     read_only: false
     volumes:
     - ".:/code"

--- a/hepdata/modules/converter/views.py
+++ b/hepdata/modules/converter/views.py
@@ -489,7 +489,7 @@ def download_datatable(datasubmission, file_format, *args, **kwargs):
         return send_file(
             dataresource.file_location,
             as_attachment=True,
-            attachment_filename=filename + '.yaml'
+            download_name=filename + '.yaml'
         )
 
     options = {
@@ -536,7 +536,7 @@ def download_datatable(datasubmission, file_format, *args, **kwargs):
         file_format = 'html'
 
     return send_file(file_to_send, as_attachment=True,
-                     attachment_filename=filename + '.' + file_format)
+                     download_name=filename + '.' + file_format)
 
 
 def display_error(title='Unknown Error', description=''):

--- a/hepdata/modules/submission/models.py
+++ b/hepdata/modules/submission/models.py
@@ -53,6 +53,9 @@ class LargeBinaryString(TypeDecorator):
     impl = types.LargeBinary
 
     def process_literal_param(self, value, dialect):
+        """
+        Encodes and returns string object.
+        """
         if isinstance(value, str):
             value = value.encode('utf-8', errors='replace')
 
@@ -61,6 +64,9 @@ class LargeBinaryString(TypeDecorator):
     process_bind_param = process_literal_param
 
     def process_result_value(self, value, dialect):
+        """
+        Encodes and returns string object.
+        """
         if isinstance(value, bytes):
             value = value.decode('utf-8', errors='replace')
 

--- a/requirements.txt
+++ b/requirements.txt
@@ -39,7 +39,7 @@ requests==2.28.1
 responses==0.22.0
 speaklater==1.3
 SQLAlchemy-Continuum==1.3.13   # Indirect: see https://github.com/inveniosoftware/invenio-records/issues/250
-SQLAlchemy==1.3.24
+SQLAlchemy==1.4.42
 SQLAlchemy-Utils==0.38.3
 timestring==1.6.4
 twitter==1.19.6

--- a/requirements.txt
+++ b/requirements.txt
@@ -5,7 +5,7 @@ celery==5.2.7
 click==8.1.3
 datacite==1.1.2
 email-validator==1.3.0
-Flask==2.0.3
+Flask==2.1.3
 Flask-CeleryExt==0.4.3
 Flask-Cors==3.0.10
 Flask-Login==0.4.1

--- a/requirements.txt
+++ b/requirements.txt
@@ -5,7 +5,7 @@ celery==5.2.7
 click==8.1.3
 datacite==1.1.2
 email-validator==1.3.0
-Flask==2.2.2
+Flask==2.0.3
 Flask-CeleryExt==0.4.3
 Flask-Cors==3.0.10
 Flask-Login==0.4.1
@@ -45,4 +45,4 @@ SQLAlchemy-Utils==0.38.3
 timestring==1.6.4
 twitter==1.19.6
 unicodeit==0.7.2
-werkzeug==2.2.2
+werkzeug==2.0.3 # Has issue with jinja v3.1.3 if updated

--- a/requirements.txt
+++ b/requirements.txt
@@ -40,7 +40,7 @@ requests==2.28.1
 responses==0.22.0
 speaklater==1.3
 SQLAlchemy-Continuum==1.3.13   # Indirect: see https://github.com/inveniosoftware/invenio-records/issues/250
-SQLAlchemy==1.3.24
+SQLAlchemy==1.4.42
 SQLAlchemy-Utils==0.38.3
 timestring==1.6.4
 twitter==1.19.6

--- a/requirements.txt
+++ b/requirements.txt
@@ -5,7 +5,7 @@ celery==5.2.7
 click==8.1.3
 datacite==1.1.2
 email-validator==1.3.0
-Flask==2.2.2
+Flask==2.0.3
 Flask-CeleryExt==0.4.3
 Flask-Cors==3.0.10
 Flask-Login==0.4.1
@@ -44,4 +44,4 @@ SQLAlchemy-Utils==0.38.3
 timestring==1.6.4
 twitter==1.19.6
 unicodeit==0.7.2
-werkzeug==2.2.2
+werkzeug==2.0.3 # Has issue with jinja v3.1.3 if updated

--- a/requirements.txt
+++ b/requirements.txt
@@ -5,7 +5,7 @@ celery==5.2.7
 click==8.1.3
 datacite==1.1.2
 email-validator==1.3.0
-Flask==2.1.3
+Flask==2.2.2
 Flask-CeleryExt==0.4.3
 Flask-Cors==3.0.10
 Flask-Login==0.4.1
@@ -44,4 +44,4 @@ SQLAlchemy-Utils==0.38.3
 timestring==1.6.4
 twitter==1.19.6
 unicodeit==0.7.2
-werkzeug==2.0.3 # Has issue with jinja v3.1.3 if updated
+werkzeug==2.2.2

--- a/requirements.txt
+++ b/requirements.txt
@@ -32,7 +32,7 @@ lxml==4.9.1
 MarkupSafe==2.1.1
 msgpack==1.0.4
 pluggy==0.13.1    # Fixed version to make sure it works with jsonresolve and pytest
-psycopg2-binary==2.9.4
+psycopg2-binary==2.9.5
 python-dateutil==2.8.2
 pyyaml==6.0
 requests==2.28.1

--- a/requirements.txt
+++ b/requirements.txt
@@ -40,7 +40,7 @@ responses==0.22.0
 speaklater==1.3
 SQLAlchemy-Continuum==1.3.13   # Indirect: see https://github.com/inveniosoftware/invenio-records/issues/250
 SQLAlchemy==1.3.24
-SQLAlchemy-Utils==0.35.0
+SQLAlchemy-Utils==0.38.3
 timestring==1.6.4
 twitter==1.19.6
 unicodeit==0.7.2

--- a/requirements.txt
+++ b/requirements.txt
@@ -41,7 +41,7 @@ responses==0.22.0
 speaklater==1.3
 SQLAlchemy-Continuum==1.3.13   # Indirect: see https://github.com/inveniosoftware/invenio-records/issues/250
 SQLAlchemy==1.3.24
-SQLAlchemy-Utils==0.35.0
+SQLAlchemy-Utils==0.38.3
 timestring==1.6.4
 twitter==1.19.6
 unicodeit==0.7.2

--- a/requirements.txt
+++ b/requirements.txt
@@ -16,7 +16,7 @@ invenio-access==1.4.4     # Indirect (needed by invenio-admin)
 invenio-accounts==1.4.13
 invenio-admin==1.3.2
 invenio-app==1.3.1 # Fixed as 1.3.4 caused error in E2E tests.
-invenio-assets==1.3.0 # Temporarily fixing as 1.3.1 causes CI error
+invenio-assets==2.0.0
 invenio-base==1.2.13
 invenio-config==1.0.3
 invenio-db==1.0.14 # v1.0.14 causes all E2E tests to fail

--- a/requirements.txt
+++ b/requirements.txt
@@ -19,7 +19,7 @@ invenio-app==1.3.1 # Fixed as 1.3.4 caused error in E2E tests.
 invenio-assets==1.3.0 # Temporarily fixing as 1.3.1 causes CI error
 invenio-base==1.2.13
 invenio-config==1.0.3
-invenio-db==1.0.13 # v1.0.14 causes all E2E tests to fail
+invenio-db==1.0.14 # v1.0.14 causes all E2E tests to fail
 invenio-logging[sentry-sdk]==1.3.2
 invenio-oauthclient==1.5.4
 invenio-pidstore==1.2.3

--- a/requirements.txt
+++ b/requirements.txt
@@ -25,7 +25,7 @@ invenio-oauthclient==1.5.4
 invenio-pidstore==1.2.3
 invenio-records==2.0.0
 invenio-search[opensearch2]==2.1.0
-invenio-theme==1.4.1
+invenio-theme==1.4.2
 invenio-userprofiles==1.2.1
 jsmin==3.0.1
 lxml==4.9.1

--- a/requirements.txt
+++ b/requirements.txt
@@ -19,7 +19,7 @@ invenio-app==1.3.1 # Fixed as 1.3.4 caused error in E2E tests.
 invenio-assets==2.0.0
 invenio-base==1.2.13
 invenio-config==1.0.3
-invenio-db==1.0.14 # v1.0.14 causes all E2E tests to fail
+invenio-db==1.0.14
 invenio-logging[sentry-sdk]==1.3.2
 invenio-oauthclient==1.5.4
 invenio-pidstore==1.2.3

--- a/requirements.txt
+++ b/requirements.txt
@@ -5,7 +5,7 @@ celery==5.2.7
 click==8.1.3
 datacite==1.1.2
 email-validator==1.3.0
-Flask==2.0.3
+Flask==2.2.2
 Flask-CeleryExt==0.4.3
 Flask-Cors==3.0.10
 Flask-Login==0.4.1
@@ -45,4 +45,4 @@ SQLAlchemy-Utils==0.35.0
 timestring==1.6.4
 twitter==1.19.6
 unicodeit==0.7.2
-werkzeug==2.0.3 # Has issue with jinja v3.1.3 if updated
+werkzeug==2.2.2

--- a/requirements.txt
+++ b/requirements.txt
@@ -5,7 +5,7 @@ celery==5.2.7
 click==8.1.3
 datacite==1.1.2
 email-validator==1.3.0
-Flask==2.0.3
+Flask==2.2.2
 Flask-CeleryExt==0.4.3
 Flask-Cors==3.0.10
 Flask-Login==0.4.1
@@ -44,4 +44,4 @@ SQLAlchemy-Utils==0.35.0
 timestring==1.6.4
 twitter==1.19.6
 unicodeit==0.7.2
-werkzeug==2.0.3 # Has issue with jinja v3.1.3 if updated
+werkzeug==2.2.2

--- a/tests/e2e/conftest.py
+++ b/tests/e2e/conftest.py
@@ -55,8 +55,10 @@ class SQLAlchemy(InvenioSQLAlchemy):
     def apply_pool_defaults(self, app, options):
         # See https://github.com/pallets/flask-sqlalchemy/issues/589#issuecomment-361075700
         # Will need updating if we move to flask-sqlalchemy >= 2.5
-        super().apply_pool_defaults(app, options)
+        options = super().apply_pool_defaults(app, options)
         options["pool_pre_ping"] = True
+        return options
+
 
 
 db = SQLAlchemy(metadata=metadata)

--- a/tests/urls_test.py
+++ b/tests/urls_test.py
@@ -29,10 +29,10 @@ from hepdata.utils.url import modify_query
 def test_url_modify(app):
     with app.app_context():
         url_path = modify_query('os_search.search', **{'date': '2001, 2002'})
-        assert(url_path == '/search/?date=2001%2C+2002')
+        assert(url_path == '/search/?date=2001,+2002')
 
         url_path = modify_query('os_search.search', **{'date': '2001, 2002', 'q': 'elastic scattering'})
-        assert(url_path == '/search/?date=2001%2C+2002&q=elastic+scattering')
+        assert(url_path == '/search/?date=2001,+2002&q=elastic+scattering')
 
         url_path = modify_query('os_search.search', **{'date': None, 'q': 'elastic scattering'})
         assert(url_path == '/search/?q=elastic+scattering')

